### DISCUSSION TITLE
Licensing road block for future contributions?

### DISCUSSION THREAD

**Discussion by henkvancann** [pandoracore/bitcoin-pro#5] [General]
This is more of a discussion than an issue.
"
License
The application is dually-licensed under AGPL v0.3 for non-commercial use and under commercial license with enterprise support by Pandora Core for commercial usage.
"
Which commercial license is it. Can't find it on the GitHub repo?
Could you elaborate on how you think this project could seduce people to contribute if you limit the application of their contribution to whatś possible within your commercial license. Should people work for free, and then you could commercially exploit the results? Why not make it fully open source and you / your projects sell your service around it?

**Comment by UkolovaOlga** [comment]
Details on the license is described [here](https://github.com/pandoracore/bitcoin-pro/blob/master/LICENSE) as in all GitHub repos
![Screenshot 2020-12-13 at 15 52 46](https://user-images.githubusercontent.com/22024706/102018731-66c85000-3d6f-11eb-9fcf-4c8f9ae08f03.png)


Answering your questions:
Open source != free. 
Bitcoin Pro is a commercial product already and is still totally open source. We cannot use non-GPL license because GTK (the UI framework that we use) demands GPL. As for the rest of the software, we support and encourage all the ideas, developments and help around Bitcoin Pro and even the reuse of the code by other companies. But as creators of the underlying technological stack ([see here](https://github.com/LNP-BP)) we are standing for fair competition and healthy free market approach, without giving away advantages instead of leveraging them.

The licenses that we chose do not limit any contributions from developers or whoever is interested in joining the project or effort. If someone is interested in becoming a part of the team - they are welcomed to join us. If anyone wants to bring more value to the product as a side contributor, whether for free, for bonuses or bounties - it also can be possible and there are no limits set from our side.

**Reply by henkvancann** [reply]
Thank you for your explanation. A few brief remarks:

a. I think I got distracted by the LICENSE file where only the open source license seems to be provided, I only saw one header there. Even https://en.wikipedia.org/wiki/GNU_Affero_General_Public_License doesn't give me a clear explanation. So I am still not sure where the commercial part of the license is hosted, which will most probable be a proprietary license "all right reserved"? Could you pinpoint or cite a (few) paragraph(s) that touches the commercial part of the dual license?

b. "Open source != free." I have been fully aware of this principle since 2005, I even build a commercial firm around it from 2006. But that doesn't answer my question about Bitcoin-Pro dual license: "could it be a future road block to contribute"?

c. I think I have my answer anyway: we are welcome to join, which is nice, but not always practical to have to settle upfront with an owner or risk the dangling investment in time and money of not having done so. 
Can it be done differently? Sure. Just as an example (not pressure there) : We've decided to split up the open source activity and commercial activity in two legal entities: Blockchainbird.org, fully open sourced, and commercial activities in Blockchain Workspace (bcws.io) ike training, projects etc.  BCWS use the results stalled under Blockchainbird, like anyone else could; frictionless, uncensored, etc.

d.  "it also can be possible and there are no limits set from our side." I.m.o. you do set limits upfront by choosing a partly commercial license.

**Comment by stefan52a** [comment]
Might I add that Bitcoin-Pro's license seems stricter (A-GPL) than that of GTK (GPL 2).
And why did you use GTK? Was RELM (MIT licensed) not good enough?

Don't get me wrong I applaud all the effort!

**Comment by dr-orlovsky** [comment]
Thnks for pointing to a good project, however as far as I see RELM is still based on GTK+ and GTK rust and just adds some syntaxic sugar on top of them - not sure it can allow to release under non-GPL license, since the software will still rely on GTK+ which is GPL
